### PR TITLE
Source map support

### PR DESCRIPTION
**Pull request by nkzawa**
Fixes https://github.com/zeit/next.js/issues/137, https://github.com/zeit/next.js/issues/225

This PR adds missing source-map support on both server and client.
But it makes the webpack compilation really slow like about +1~2 seconds on HMR.
I doubt if we should make this default yet or can make it faster somehow.

### CODE REVIEW

File: server/build/webpack.js [hunk 1]
```diff
@@ -178,6 +189,7 @@ export default async function createCompiler (dir, { hotReload = false, dev = fa
       ],
       loaders
     },
+    devtool: dev ? 'inline-source-map' : false,
```
Comment (nkzawa) [inline server/build/webpack.js]: Some other `devtool` settings is faster, but didn't properly work as far as I tested.

**Comment by arunoda** [comment]
About speeding up, have we enabled cacheDirectory setting of the babel. It might give us some speed benefits in the long run.

File: server/build/webpack.js (hunk 1 above)
Comment (jonaswindey) [inline server/build/webpack.js]: I tested with `cheap-module-eval-source-map` and for me this is the fastest setting with correct breakpoints (Chrome on OS X)

Comment (rauchg) [inline server/build/webpack.js]: @jonaswindey can you confirm if it works for you with correct source locations?

**Review by rauchg — APPROVED** [review]

File: server/build/webpack.js (hunk 1 above)
Comment (rauchg) [inline server/build/webpack.js]: @jonaswindey I'm merging, but if there's a faster alternative that works, please let us know

**Review by rauchg — APPROVED** [review]

**Comment by nkzawa** [comment]
It seems this works faster than before I tested, don't know why though 🤔

**Comment by arunoda** [comment]
@nkzawa Interesting :)

**Pull request merged**